[PyTorch] Remove hardcoded device=cuda in test_aot_inductor

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #113577
* #112800
* #112799
* #112798
* #112405
* #112174
* #112116
* #112115
* __->__ #112797
* #113997

All the other tests use self.device, so this seems like an oversight? Cost me a lot of time debugging the minimal arrayref interface, which is only intended for CPU.

Differential Revision: [D50949928](https://our.internmc.facebook.com/intern/diff/D50949928/)

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

swolchok committed:
[PyTorch] Remove hardcoded device=cuda in test_aot_inductor

All the other tests use self.device, so this seems like an oversight? Cost me a lot of time debugging the minimal arrayref interface, which is only intended for CPU.

Differential Revision: [D50949928](https://our.internmc.facebook.com/intern/diff/D50949928/)

[ghstack-poisoned]

chenyang78 reviewed (approved):
LGTM. Thanks.

swolchok commented:
@pytorchbot merge

desertfire commented on test/inductor/test_aot_inductor.py:
```diff
@@ -710,8 +710,8 @@ def forward(self, x, y):
-            torch.tensor([1, 1, 1], device="cuda"),
-            torch.randn((1, 32), dtype=torch.float16, device="cuda"),
+            torch.tensor([1, 1, 1], device=self.device),
+            torch.randn((1, 32), dtype=torch.float16, device=self.device),
```
You can use 
```
        if self.device != "cuda":
            raise unittest.SkipTest("requires CUDA")
```
to skip this test since float16 won't work for cpu (see the CI failure msg).

swolchok committed:
Update on "[PyTorch] Remove hardcoded device=cuda in test_aot_inductor"


All the other tests use self.device, so this seems like an oversight? Cost me a lot of time debugging the minimal arrayref interface, which is only intended for CPU.

Differential Revision: [D50949928](https://our.internmc.facebook.com/intern/diff/D50949928/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Remove hardcoded device=cuda in test_aot_inductor"


All the other tests use self.device, so this seems like an oversight? Cost me a lot of time debugging the minimal arrayref interface, which is only intended for CPU.

Differential Revision: [D50949928](https://our.internmc.facebook.com/intern/diff/D50949928/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Remove hardcoded device=cuda in test_aot_inductor"


All the other tests use self.device, so this seems like an oversight? Cost me a lot of time debugging the minimal arrayref interface, which is only intended for CPU.

Differential Revision: [D50949928](https://our.internmc.facebook.com/intern/diff/D50949928/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Remove hardcoded device=cuda in test_aot_inductor"


All the other tests use self.device, so this seems like an oversight? Cost me a lot of time debugging the minimal arrayref interface, which is only intended for CPU.

Differential Revision: [D50949928](https://our.internmc.facebook.com/intern/diff/D50949928/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Remove hardcoded device=cuda in test_aot_inductor"


All the other tests use self.device, so this seems like an oversight? Cost me a lot of time debugging the minimal arrayref interface, which is only intended for CPU.

Differential Revision: [D50949928](https://our.internmc.facebook.com/intern/diff/D50949928/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Remove hardcoded device=cuda in test_aot_inductor"


All the other tests use self.device, so this seems like an oversight? Cost me a lot of time debugging the minimal arrayref interface, which is only intended for CPU.

Differential Revision: [D50949928](https://our.internmc.facebook.com/intern/diff/D50949928/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Remove hardcoded device=cuda in test_aot_inductor"


All the other tests use self.device, so this seems like an oversight? Cost me a lot of time debugging the minimal arrayref interface, which is only intended for CPU.

Differential Revision: [D50949928](https://our.internmc.facebook.com/intern/diff/D50949928/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Remove hardcoded device=cuda in test_aot_inductor"


All the other tests use self.device, so this seems like an oversight? Cost me a lot of time debugging the minimal arrayref interface, which is only intended for CPU.

Differential Revision: [D50949928](https://our.internmc.facebook.com/intern/diff/D50949928/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Remove hardcoded device=cuda in test_aot_inductor"


All the other tests use self.device, so this seems like an oversight? Cost me a lot of time debugging the minimal arrayref interface, which is only intended for CPU.

Differential Revision: [D50949928](https://our.internmc.facebook.com/intern/diff/D50949928/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Remove hardcoded device=cuda in test_aot_inductor"


All the other tests use self.device, so this seems like an oversight? Cost me a lot of time debugging the minimal arrayref interface, which is only intended for CPU.

Differential Revision: [D50949928](https://our.internmc.facebook.com/intern/diff/D50949928/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Remove hardcoded device=cuda in test_aot_inductor"


All the other tests use self.device, so this seems like an oversight? Cost me a lot of time debugging the minimal arrayref interface, which is only intended for CPU.

Differential Revision: [D50949928](https://our.internmc.facebook.com/intern/diff/D50949928/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

swolchok commented:
@pytorchbot merge
